biofefi/components/forms.py

```diff
@@ -543,51 +544,107 @@ def pairplot_form(data, data_analysis_plot_dir, plot_opts):
 
 
 @st.experimental_fragment
-def tSNE_plot_form(data, random_state, data_analysis_plot_dir, plot_opts):
+def tSNE_plot_form(data, random_state, data_analysis_plot_dir, plot_opts, scaler=None):
 
     from sklearn.manifold import TSNE
-    from sklearn.preprocessing import StandardScaler
+    from sklearn.preprocessing import MinMaxScaler, StandardScaler
```

Comment:
How come you're doing the import inside the function instead of at the top of the file?